datafusion/core/src/dataframe/mod.rs

```diff
@@ -2869,6 +2876,39 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn test_window_function_with_column() -> Result<()> {
+        let df = test_table().await?.select_columns(&["c1", "c2", "c3"])?;
+        let ctx = SessionContext::new();
+        let df_impl = DataFrame::new(ctx.state(), df.plan.clone());
+        let func = Expr::WindowFunction(WindowFunction::new(
+            WindowFunctionDefinition::BuiltInWindowFunction(
+                BuiltInWindowFunction::RowNumber,
+            ),
+            vec![],
+        ))
+        .alias("row_num");
```

Comment:
I think you can use the expr fn here and make this more concise:



```suggestion
        let func = row_number().alias("row_num");
```

Reply:
I also ran this test without the code changes and it fails like this:

```

assertion `left == right` failed
  left: 4
 right: 5

Left:  4
Right: 5
<Click to see difference>

thread 'dataframe::tests::test_window_function_with_column' panicked at datafusion/core/src/dataframe/mod.rs:2882:9:
assertion `left == right` failed
  left: 4
 right: 5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/3f5fd8dd41153bc5fdca9427e9e05be2c767ba23/library/std/src/panicking.rs:652:5
...

```

And the output was like

```
[
    "+----+----+-----+-----------------------------------------------------------------------+---+",
    "| c1 | c2 | c3  | ROW_NUMBER() ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING | r |",
    "+----+----+-----+-----------------------------------------------------------------------+---+",
    "| c  | 2  | 1   | 1                                                                     | 1 |",
    "| d  | 5  | -40 | 2                                                                     | 2 |",
    "+----+----+-----+-----------------------------------------------------------------------+---+",
]
```

Reply:
Sounds good I went ahead and used the more concise method call. Thanks!